keps/487-kubectl-plugin-listing/README.md

```diff
@@ -0,0 +1,316 @@
+      Spec:
+        Containers:
+          Args:
+            10s
+          Image:              gcr.io/k8s-staging-perf-tests/sleep:latest
+          Image Pull Policy:  Always
+          Name:               dummy-job
+          Resources:
+            Limits:
+              Cpu:                  500m
+              Ephemeral - Storage:  512Mi
+              Memory:               512Mi
+              nvidia.com/gpu:       1
+            Requests:
+              Cpu:                     500m
+              Ephemeral - Storage:     512Mi
+              Memory:                  512Mi
+              nvidia.com/gpu:          1
+          Termination Message Path:    /dev/termination-log
+          Termination Message Policy:  File
+        Dns Policy:                    ClusterFirst
+        Node Selector:
+          cloud.google.com/gke-accelerator:  nvidia-tesla-t4
+        Restart Policy:                      Never
+        Scheduler Name:                      default-scheduler
+        Security Context:
+        Termination Grace Period Seconds:  30
+  Priority:                                0
+  Priority Class Source:                   
+  Queue Name:                              lq-team-a
+
+```
+
+In order to make kueue CLI simple, we are going to have only one abstraction for jobs specific information — workload. Currently the `describe` command will mostly duplicate the `kubectl describe` command with few modifications:
+`--status` option will display Workload.Status field and state of Workload.
+There is no `State` field in Workload, but it could be identified from other fields. The `Workload.Status` has an `Admitted` field, in case of pending workload it is empty, so from the user point of view it's not clear what the state of the Workload is. I suggest adding the state information that displays `Admitted` if the `Status.Admitted` is not empty and `Pending` if the `Status.Admitted` is empty.
+The `--spec` option will display `Workload.Spec` information.
+
+For ClusterQueue the Status field might look confusing. For example for misconfigured ClusterQueur `Status.Conditions[-1].Type` have value `Active` and `Status.Conditions[-1].Status` have value `False`. I suggest including a state field that will display `Error` if `Status.Conditions.Status` is False and `Active` if `Status.Conditions.Status` is `True` and `Status.Conditions.Type` is `Active`. Later, when we'll support other types, the logic will probably stay the same, but it should be double checked.
+
+Since the logic for the `State` field is complex, we need to make sure it's well tested and be able to identify when the logic should be updated. The proposal is to have a dedicated test for `State` next to the API definition.
+
+### List
+
+```bash
+$kubectl kueue worlkoads list --clusterqueue=cq
+| Namespace | Name     |  Queue  | ClusterQueue | Pods |   State   | Age
+|-----------|----------|---------|--------------|------|-----------|-----
```

Comment:
Can we skip table drawing and have it in the same way as kubectl does? Drawing tables make it a bit inconvenient to use kueuectl output as a input to other command.